src/CoreWCF.Primitives/src/CoreWCF.Primitives.csproj

```diff
@@ -1,12 +1,40 @@
 <Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>$(LibraryTargetFrameworks)</TargetFrameworks>
     <AssemblyName>CoreWCF.Primitives</AssemblyName>
     <PackageId>CoreWCF.Primitives</PackageId>
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
     <RootNamespace />
   </PropertyGroup>
+
   <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.1" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)'=='net6.0' or $(IsAspNetCore) != true">
+    <PackageReference Include="Microsoft.IdentityModel.Protocols.WsFederation" Version="6.35.1" />
+    <PackageReference Include="Microsoft.IdentityModel.Tokens" Version="6.35.1" />
+    <PackageReference Include="Microsoft.IdentityModel.Tokens.Saml" Version="6.35.1" />
+    <PackageReference Include="System.DirectoryServices" Version="6.0.1" />
+    <PackageReference Include="System.DirectoryServices.Protocols" Version="6.0.2" />
+    <PackageReference Include="System.Security.Cryptography.ProtectedData" Version="6.0.0" />
+    <PackageReference Include="System.Security.Cryptography.Xml" Version="6.0.1" />
+    <PackageReference Include="System.Web.Services.Description" Version="6.2.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)'=='net8.0'">
```

Comment:
reduce the dependencies introduced for .NET 6 and .NET 8.

The test file src/CoreWCF.BuildTools/tests/ReferenceAssembliesHelper.cs needs to match these ItemGroup conditions to correctly load the referenced dependencies, otherwise tests fail